agent: backhaul_manager: fix remove_client_from_all_radios

arnout opened the pull request:
Fixes #1241.

Include a test.
It doesn't actually test the functionality of that function (we'd need to parse topology response for that), but it does trigger the crash from #1241.

arnout committed:
agent: backhaul_manager: remove_client_from_all_radios: remove client

remove_client_from_all_radios() should remove the client from the
associated_clients_map. Instead, it was removing the slave socket. Fix
this.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
tests: connmap: parse clients as well

Include the clients in the parsed connection map. Add a new class
MapClient for it.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
test_flows.py: client_association_dummy: extend test with conn_map

Extend the test to verify that the connection map is updated correctly.
Also test that the client is removed if there is no explicit
disassociate.

Incidentally, this triggers bug #1241 so it can serve as a regression
test.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

tomereli commented on tests/connmap.py:
```diff
@@ -14,12 +14,24 @@
+    def __init__(self, mac: str):
+        self.mac = mac
+
+
```
Extra line break - doesn't flake8 complain about this?

arnout replied:
I don't see the extra line break... PEP8 specifies two empty lines before each top-level class or function definition.

tomereli commented:
@itayx please confirm that this resolves the issue.

itayx committed:
bwl: fix missing argument

As part of the STA_Connected event we load the station information from
the dwpal.

Fix the index argument in sClientAssociationParams.
Add a break in case one of the params in invalid.

Signed-off-by: Itay Elenzweig <itayx.elenzweig@intel.com>

itayx commented:
> @itayx please confirm that this resolves the issue.

This fixes the client association described in #1241 
In addition I added a fix that was discussed in the Slack #prplmesh-dev
https://prplfoundation.slack.com/archives/CKKLX17P1/p1588845700382900

arnout commented:
Itay's additional fix LGTM, so merging.
